### PR TITLE
[Flow] Raise special `linalg.generic` ops to `linalg.fill` ops

### PR DESCRIPTION
**Pull request by Max191**

### CODE REVIEW

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/RaiseSpecialOps.cpp [hunk 1]
```diff
@@ -76,6 +76,26 @@ std::optional<Value> matchATransposeBMatmul(linalg::LinalgOp matmulOp) {
...
+std::optional<Value> matchGenericFill(linalg::LinalgOp linalgOp) {
+  if (isa<linalg::GenericOp>(linalgOp.getOperation()) &&
+      linalgOp.getNumDpsInputs() == 0 && linalgOp.getNumDpsInits() == 1 &&
+      linalgOp.getNumParallelLoops() == linalgOp.getNumLoops() &&
+      linalgOp.getIndexingMapsArray()[0].isIdentity()) {
+    // Check that the op body is only a linalg.yield op.
+    Value yieldOperand;
+    for (Operation &bodyOp : linalgOp.getBlock()->getOperations()) {
+      if (isa<linalg::YieldOp>(bodyOp))
+        yieldOperand = bodyOp.getOperand(0);
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/RaiseSpecialOps.cpp:90]: You also need to check that the yield operand is not the basic block argument of the linalg operation itself.

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]
Nice! Just one minor comment.

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/RaiseSpecialOps.cpp (hunk 1 above)
Comment (Max191) [reply compiler/src/iree/compiler/Dialect/Flow/Transforms/RaiseSpecialOps.cpp:90]: Should be good now. Lmk what you think.

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/RaiseSpecialOps.cpp [hunk 2]
```diff
@@ -76,6 +76,31 @@ std::optional<Value> matchATransposeBMatmul(linalg::LinalgOp matmulOp) {
...
+// the fill value (input operand to linalg.fill) on success.
+std::optional<Value> matchGenericFill(linalg::LinalgOp linalgOp) {
+  if (isa<linalg::GenericOp>(linalgOp.getOperation()) &&
+      linalgOp.getNumDpsInputs() == 0 && linalgOp.getNumDpsInits() == 1 &&
+      linalgOp.getNumParallelLoops() == linalgOp.getNumLoops() &&
+      linalgOp.getIndexingMapsArray()[0].isIdentity()) {
+    // Check that the op body is only a linalg.yield op.
+    Value yieldOperand;
+    for (Operation &bodyOp : linalgOp.getBlock()->getOperations()) {
+      if (isa<linalg::YieldOp>(bodyOp))
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/RaiseSpecialOps.cpp]: Nit: please add `{` around all statements (this is divergence in IREE style w.r.t LLVM style)

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/RaiseSpecialOps.cpp [hunk 3]
```diff
@@ -76,6 +76,31 @@ std::optional<Value> matchATransposeBMatmul(linalg::LinalgOp matmulOp) {
...
+        yieldOperand = bodyOp.getOperand(0);
+      else
+        return std::nullopt;
+    }
+    // Check that the operand of the linalg.yield op is not an argument of the
+    // linalg.generic basic block
+    for (Value blockArg : linalgOp.getBlock()->getArguments())
+      if (yieldOperand == blockArg)
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/RaiseSpecialOps.cpp]: Same nit here. Please add braces always in IREE.

**Review by MaheshRavishankar — APPROVED** [review]
Sorry, forgot to push my comments from yesterday. I think this works. Left a few comments. There could be a cleanup of the logic that could be done after the fact to simplify the code. Happy to chat more.

**Comment by Max191** [comment]
rebased on top of main

**Pull request merged**